### PR TITLE
perf: de-select unnecessary columns

### PR DESCRIPTION
**Pull request by gtarpenning**
https://wandb.atlassian.net/browse/WB-20367

This pr: 
- explicitly selects columns from the backend (which de-selects costly input/output fields)
- the result **should** be much speedier loading in the detail view, and much less data transferred over the wire, with a tradeoff that it will be slower when clicking between items in the tree

Depends on this core fix: https://github.com/wandb/core/pull/23560

In master, loading this trace (peek) takes 3s and loads `26MB` in this [proj](https://wandb.ai/geekyrakshit/prompt-upsampling-diffusion/weave/traces?peekPath=%2Fgeekyrakshit%2Fprompt-upsampling-diffusion%2Fcalls%2Fa58440e9-f952-444e-81aa-31560f54eea0%3Fpath%3Ddspy.Module*0%2BMultiModalJudgeModule.forward*0%26tracetree%3D1)
Screenshot 2024-08-22 at 1 52 40 PM

In this branch, loading the [same proj](https://wandb.ai/geekyrakshit/prompt-upsampling-diffusion/weave/traces?peekPath=%2Fgeekyrakshit%2Fprompt-upsampling-diffusion%2Fcalls%2Fa58440e9-f952-444e-81aa-31560f54eea0%3Fpath%3Ddspy.Module*0%2BMultiModalJudgeModule.forward*0%26tracetree%3D1) detail trace tree loads `5.5KB` in 800ms
Screenshot 2024-08-22 at 1 52 09 PM

In larger projects with huge input/output shapes, this discrepancy is much larger

### CODE REVIEW

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooks.ts [hunk 1]
```diff
@@ -325,22 +328,24 @@ const useCallsNoExpansion = (
...
-        callCache.set(
-          {
-            entity,
-            project,
-            callId: call.callId,
-          },
-          call
-        );
-      });
+      if (!columns) {
```
Comment (gtarpenning) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooks.ts:331]: we can't cache results if the query had a specific column request, as future queries might not have the same columns. We could add the columns to the cache key, but then we invalidate our deletion from cache scheme. Almost always (currently), specifying columns makes the query fast, so caching is less important. Not a longterm solution.

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallPage.tsx [hunk 2]
```diff
@@ -124,16 +125,27 @@ const CallPageInnerVertical: FC<{
...
   const {rows, expandKeys, loading} = tree;
-  let {selectedCall} = tree;
+
+  const {selectedCall} = tree;
+
+  const callComplete = useCall({
+    entity: selectedCall.entity,
+    project: selectedCall.project,
+    callId: selectedCall.callId,
+  });
```
Comment (gtarpenning) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallPage.tsx:135]: query for full-fat call on selection change

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallDetails.tsx [hunk 3]
```diff
@@ -258,7 +279,9 @@ const getDisplayInputsAndOutput = (call: CallSchema) => {
   const span = call.rawSpan;
   const inputKeys =
     span.inputs._keys ??
-    Object.keys(span.inputs).filter(k => !k.startsWith('_') || k === '_type');
+    Object.keys(span.inputs ?? {}).filter(
```
Comment (tssweeney) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallDetails.tsx:282]: shouldn't inputs be empty dict even when we don't select it?

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooks.ts (hunk 1 above)
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooks.ts:337]: very good! i was wondering if this PR would account for that - nice (:

Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/wfReactInterface/tsDataModelHooks.ts:337]: would be nice to add this as a comment

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallDetails.tsx (hunk 3 above)
Comment (gtarpenning) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallDetails.tsx:282]: er, yes, but currently we rely on the request columns entirely. For **all** fields. We would have to add some required column logic in the stream builder.

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallTraceView.tsx [hunk 4]
```diff
@@ -249,6 +249,9 @@ const RE_TRAILING_INT = /\d+$/;
 // Sorting evaluation calls by dataset row.
 // Because of async they may have been processed in a different order.
 const getCallSortExampleRow = (call: CallSchema): number => {
+  if (!call.rawSpan.inputs?._keys) {
```
Comment (tssweeney) [inline weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallTraceView.tsx:252]: `_keys` is not right here - not sure why this is a check?

Comment (gtarpenning) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallTraceView.tsx:252]: ?? dev tracks I think. Sloppy

File: weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallDetails.tsx (hunk 3 above)
Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallDetails.tsx:282]: so this is a bug in the server?

Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallDetails.tsx:282]: the return type of `inputs` from the server is always a dictionary

Comment (tssweeney) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallDetails.tsx:282]: or it should be

Comment (gtarpenning) [reply weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallPage/CallDetails.tsx:282]: Okay this is pretty dumb but essentially what is required: https://github.com/wandb/core/pull/23508. @tssweeney

**Review by tssweeney — APPROVED** [review]

**Pull request merged**